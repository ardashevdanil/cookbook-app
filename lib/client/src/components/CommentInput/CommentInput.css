.CommentInput {
  box-sizing: border-box;
  margin: 45px 0;
  padding: 20px 40px 30px 40px;
  background-color: white;
  box-shadow: 0 0 13px 0 rgba(11, 12, 12, 0.15);
}

.CommentInput__heading {
  font-size: 24px;
  margin-bottom: 16px;
}

.CommentInput__input {
  padding: 26px;
  display: flex;
  align-items: center;
  background-color: #f6f6f6;
}

.CommentInput__input>input {
  width: calc(100% - 85px - 7px);
  height: 38px;
  border: solid 1px #c2c2c2;
  background-color: white;
}
  
.CommentInput__button {
  width: 85px;
  height: 38px;
  margin-left: 7px;
  border: 1px solid #383737;
  background-color: #383737;
  color: white;
  font-size: 17px;
  display: flex;
  align-items: center;
  justify-content: center;
}

.CommentInput__button:hover {
  cursor: pointer;
  color: #383737;
  background-color: white;
}

@media only screen and (max-width: 1120px) {
  .CommentInput {
    margin: 0;
    padding: 7vw 6.9vw;
    background-color: #f6f6f6;
    box-shadow: none;
  }

  .CommentInput__heading {
    font-size: 7vw;
  }

  .CommentInput__input {
    flex-direction: column;
    padding: 0;
  }

  .CommentInput__input>input {
    width: 100%;
  }

  .CommentInput__button {
    width: 100%;
    margin: 0;
    font-size: 5.5vw;
    margin-top: 1vw;
  }
}
