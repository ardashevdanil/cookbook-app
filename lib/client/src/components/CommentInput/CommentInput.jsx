import axios from 'axios';
import React from 'react';
import './CommentInput.css';

class CommentInput extends React.Component {
  constructor() {
    super();
    this.state = {
      value: '',
    };
    this.handleChange = this.handleChange.bind(this);
    this.postComment = this.postComment.bind(this);
  }

  handleChange(e) {
    this.setState({
      value: e.target.value,
    });
  }

  postComment() {
    const { value } = { ...this.state };
    const { recipe, user } = { ...this.props };

    if (!value) return;

    axios.post('/comments', {
      recipeId: recipe,
      text: value,
      userName: user,
    });
    this.setState({ value: '' });
  }

  render() {
    const { value } = { ...this.state };

    return (
      <div className="CommentInput">
        <div className="CommentInput__heading">
          Post a Comment
        </div>
        <div className="CommentInput__input">
          <input
            onChange={this.handleChange}
            type="text"
            value={value}
          />
          <div
            className="CommentInput__button"
            onClick={this.postComment}
            role="button"
          >
            POST
          </div>
        </div>
      </div>
    );
  }
}

export default CommentInput;
