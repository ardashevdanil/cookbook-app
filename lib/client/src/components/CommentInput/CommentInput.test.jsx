import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';

import CommentInput from './CommentInput';

Enzyme.configure({ adapter: new Adapter() });

describe('test CommentInput component', () => {
  const component = Enzyme.shallow(
    <CommentInput />,
  );

  test('renders correctly', () => {
    expect(component.debug()).toMatchSnapshot();
  });
});
