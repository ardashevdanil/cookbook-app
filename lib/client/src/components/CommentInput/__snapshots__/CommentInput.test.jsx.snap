// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test CommentInput component renders correctly 1`] = `
"<div className=\\"CommentInput\\">
  <div className=\\"CommentInput__heading\\">
    Post a Comment
  </div>
  <div className=\\"CommentInput__input\\">
    <input onChange={[Function: bound handleChange]} type=\\"text\\" value=\\"\\" />
    <div className=\\"CommentInput__button\\" onClick={[Function: bound postComment]} role=\\"button\\">
      POST
    </div>
  </div>
</div>"
`;
